### PR TITLE
Processing - Add GeoPackage support in alg qgis:convertformat

### PR DESCRIPTION
**Pull request by mdouchin**
## Description

The alg qgis;convertformat allows to use ogr2ogr in Processing, but lacks the support of GeoPackage in QGIS 2.18. This commit adds the support for GeoPackage format as output format in the qgis:convertformat python script.

I think this can be considered as a bug fix ?

### CODE REVIEW

**Review by rldhont — APPROVED** [review]

**Comment by luipir** [comment]
before to merge @mdouchin i would ask to:
1) this affect also master? need a forward-port?
2) would be fine to create test for this feature as described in https://github.com/qgis/QGIS/tree/release-2_18/python/plugins/processing/tests and forward port to master.
The rationale is that every one has a verifiable processing workflow can add his little test. It does not cost so much work.

**Comment by nyalldawson** [comment]
Actually in this case you're better off backporting the tests from https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/GdalAlgorithmsTest.py#L1387 and adding your new test there.

**Comment by mdouchin** [comment]
Thanks for your review. I will try to improve master tests and then backport here

**Comment by luipir** [comment]
great and thanks @mdouchin it is really useful to cover any kind of parameter test. This will avoid regressions due to backend version changes

File: python/plugins/processing/tests/GdalAlgorithmsTest.py [hunk 1]
```diff
@@ -54,6 +57,28 @@ def tearDownClass(cls):
     def test_definition_file(self):
         return 'gdal_algorithm_tests.yaml'
 
+    def testOgr2Ogr(self):
+        source = os.path.join(testDataPath, 'polys.gml')
+
+        alg = Ogr2Ogr()
+        alg.setParameterValue('INPUT_LAYER', source)
```
Comment (rldhont) [inline python/plugins/processing/tests/GdalAlgorithmsTest.py]: @mdouchin can you add this line ?
```
alg.setParameterValue('FORMAT', 0)
```

Comment (mdouchin) [reply python/plugins/processing/tests/GdalAlgorithmsTest.py]: done !

File: python/plugins/processing/tests/GdalAlgorithmsTest.py [hunk 2]
```diff
@@ -54,6 +57,29 @@ def tearDownClass(cls):
...
+        alg.setParameterValue('FORMAT', 0)
+        alg.setOutputValue('OUTPUT_LAYER', 'd:/temp/check.shp')
+        self.assertEqual(
+            alg.getConsoleCommands(),
+            ['ogr2ogr',
+             '-f "ESRI Shapefile" d:/temp/check.shp ' +
```
Comment (rldhont) [inline python/plugins/processing/tests/GdalAlgorithmsTest.py]:
```suggestion
             '-f "ESRI Shapefile" "d:/temp/check.shp" ' +
```

File: python/plugins/processing/tests/GdalAlgorithmsTest.py [hunk 3]
```diff
@@ -54,6 +57,29 @@ def tearDownClass(cls):
...
+             source + ' polys2'])
+
+        alg = Ogr2Ogr()
+        alg.setParameterValue('INPUT_LAYER', source)
+        alg.setParameterValue('FORMAT', 22)
+        alg.setOutputValue('OUTPUT_LAYER', 'd:/temp/check.gpkg')
+        self.assertEqual(
+            alg.getConsoleCommands(),
+            ['ogr2ogr',
+             '-f "GPKG" d:/temp/check.shp ' +
```
Comment (rldhont) [inline python/plugins/processing/tests/GdalAlgorithmsTest.py]:
```suggestion
             '-f "GPKG" "d:/temp/check.shp" ' +
```

File: python/plugins/processing/tests/GdalAlgorithmsTest.py [hunk 4]
```diff
@@ -54,6 +57,29 @@ def tearDownClass(cls):
...
+             source + ' polys2'])
+
+        alg = Ogr2Ogr()
+        alg.setParameterValue('INPUT_LAYER', source)
+        alg.setParameterValue('FORMAT', 22)
+        alg.setOutputValue('OUTPUT_LAYER', 'd:/temp/check.gpkg')
+        self.assertEqual(
+            alg.getConsoleCommands(),
+            ['ogr2ogr',
+             '-f "GPKG" "d:/temp/check.shp" ' +
```
Comment (rldhont) [inline python/plugins/processing/tests/GdalAlgorithmsTest.py]:
```suggestion
             '-f GPKG "d:/temp/check.gpkg" ' +
```

**Comment by rldhont** [comment]
Forward porting done!

**Pull request merged**